lib/ansible/modules/remote_management/dellemc/ome/ome_device_info.py

```diff
@@ -0,0 +1,418 @@
+        raise ValueError("One of the following {0} is required for {1}".format(required_args, val))
+
+
+def _validate_inputs(module_params):
+    """validates input parameters"""
+    fact_subset = module_params["fact_subset"]
+    if fact_subset != "basic_inventory":
+        _check_mutually_inclusive_arguments(fact_subset, module_params, ["device_id", "device_service_tag"])
+
+
+def main():
+    system_query_options = {"type": 'dict', "required": False, "options": {
+        "device_id": {"type": 'list'},
+        "device_service_tag": {"type": 'list'},
+        "inventory_type": {"type": 'str'},
+        "filter": {"type": 'str', "required": False},
+    }}
+
+    module = AnsibleModule(
+        argument_spec={
+            "hostname": {"required": True, "type": 'str'},
+            "username": {"required": True, "type": 'str'},
+            "password": {"required": True, "type": 'str', "no_log": True},
+            "port": {"required": False, "default": 443, "type": 'int'},
+            "fact_subset": {"required": False, "default": "basic_inventory",
+                            "choices": ['basic_inventory', 'detailed_inventory', 'subsystem_health']},
+            "system_query_options": system_query_options,
+        },
+        required_if=[['fact_subset', 'detailed_inventory', ['system_query_options']],
+                     ['fact_subset', 'subsystem_health', ['system_query_options']], ],
+        supports_check_mode=False)
+
+    try:
+        _validate_inputs(module.params)
+        with RestOME(module.params, req_session=True) as rest_obj:
+            device_facts = _get_resource_parameters(module.params, rest_obj)
+            resp_status = []
+            if device_facts.get("basic_inventory"):
+                query_param = _get_query_parameters(module.params)
+                resp = rest_obj.invoke_request('GET', device_facts["basic_inventory"], query_param=query_param)
+                device_facts = resp.json_data
+                resp_status.append(resp.status_code)
+            else:
+                for identifier_type, path_dict_map in device_facts.items():
+                    for identifier, path in path_dict_map.items():
+                        try:
+                            resp = rest_obj.invoke_request('GET', path)
+                            data = resp.json_data
+                            resp_status.append(resp.status_code)
+                        except HTTPError as err:
+                            data = str(err)
+                        path_dict_map[identifier] = data
+                if any(device_fact_error_report):
+                    if "device_service_tag" in device_facts:
+                        device_facts["device_service_tag"].update(device_fact_error_report)
+                    else:
+                        device_facts["device_service_tag"] = device_fact_error_report
+        if 200 in resp_status:
+            module.exit_json(device_info= device_facts)
```

Comment:
```suggestion
            module.exit_json(device_info=device_facts)
```
Some of the sanity checks are failing because of this space. (ansibot should post the sanity test results, but apparently it isn't...)

Reply:
@felixfontein thanks for the review. I will update the changes and update now